ydb/core/persqueue/partition.cpp

```diff
@@ -364,6 +362,24 @@ void TPartition::AddMetaKey(TEvKeyValue::TEvRequest* request) {
     meta.SetEndOffset(Max(NewHead.GetNextOffset(), EndOffset));
     meta.SetSubDomainOutOfSpace(SubDomainOutOfSpace);
 
+    if (IsSupportive()) {
+        auto* counterData = meta.MutableCounterData();
+        counterData->SetMessagesWrittenGrpc(MsgsWrittenGrpc.Value());
+        counterData->SetMessagesWrittenTotal(MsgsWrittenTotal.Value());
+        counterData->SetBytesWrittenGrpc(BytesWrittenGrpc.Value());
+        counterData->SetBytesWrittenTotal(BytesWrittenTotal.Value());
+        counterData->SetBytesWrittenUncompressed(BytesWrittenUncompressed.Value());
+        for(const auto& v : MessageSize.GetValues()) {
+            counterData->AddMessagesSizes(v);
+        }
+        for(const auto& v : PartitionWriteQuotaWaitCounter.GetValues()) {
+            counterData->AddPartitionWriteQuotaWait(v);
```

Comment:
Do not count this counters on commit of tx. This must be counted during write to supportive partition.